File: python/test/inferer.py
#!/usr/bin/env python

"""
:author Kier
"""

from typing import Union, List, Dict

import tensorstack as ts

if __name__ == "__main__":
    model = "/home/kier/git/TensorStack/python/test/caffe.tsm"

    with open(model, "rb") as f:
        module = ts.module.Module.Load(f)

    outputs = module.outputs

    for i in module.inputs:
        if not ts.inferer.has_infered(i):
            if not i.has(ts.Node.RetentionParam.dtype):
                i.dtype = ts.FLOAT32
            if not i.has(ts.Node.RetentionParam.shape):
                i.shape = [-1, 3, 256, 256]

    print(outputs[0])
    print(ts.inferer.infer(outputs[0]))
    print(outputs[0])

File: test/inferer.cpp
//
// Created by kier on 2019/11/22.
//

#include "runtime/inferer.h"
#include "module/module.h"
#include "core/tensor_builder.h"

int main() {
    std::string module_path = "/home/kier/git/TensorStack/python/test/caffe.tsm";

    auto module = ts::Module::Load(module_path);

    auto input = module->input(0);
    if (!input->has("#shape"))
        input->set("#shape", ts::tensor::build(ts::INT32, {-1, 256, 256, 3}));

    auto output = module->output(0);

    auto output_proto = ts::infer(output);

    TS_LOG_INFO << output_proto;

    return 0;
}